Remove buildpaths

Remove the hardcoded buildpaths by using the Eigen3 component.

Upstream-Status: Pending

Signed-off-by: Rob Woolley <rob.woolley@windriver.com>

Index: git/CMakeLists.txt
===================================================================
--- git.orig/CMakeLists.txt
+++ git/CMakeLists.txt
@@ -62,6 +62,7 @@ if(BUILD_TESTING AND ament_cmake_catch2_
       rmf_traffic
       ${FCL_LIBRARIES}
       Threads::Threads
+      Eigen3::Eigen
   )
 
   if(using_new_fcl)
@@ -69,8 +70,6 @@ if(BUILD_TESTING AND ament_cmake_catch2_
   endif()
 
   target_include_directories(test_rmf_traffic
-    PUBLIC
-      "${Eigen3_INCLUDE_DIRS}"
     PRIVATE
       $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/>
   )
@@ -98,6 +97,7 @@ target_link_libraries(rmf_traffic
   PUBLIC
     rmf_utils::rmf_utils
     Threads::Threads
+    Eigen3::Eigen
   PRIVATE
     ${FCL_LIBRARIES}
 )
@@ -106,7 +106,6 @@ target_include_directories(rmf_traffic
   PUBLIC
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
     $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
-    ${EIGEN3_INCLUDE_DIRS}
 )
 
 # Create cmake config files
